RFC-0011-open-telemetry.md

```diff
@@ -0,0 +1,234 @@
+# **RFC0011 for Presto**
+
+
+## Enhancing Open Telemetry Implementation in Presto
+Proposers
+* Suresh Babu Areekara
+* Siddarth Ajay
+* Ben Tony Joe
+
+
+## [Related Issues]
+* https://github.com/prestodb/presto/issues/23975
+
+
+## Summary
+The existing Open Telemetry implementation https://github.com/prestodb/presto/pull/18534 was an experimental feature, had a limited set of telemetry data(Query state changes) and did not include a child span concept. The recent implementation will make Presto more flexible, allowing support for both parent and child spans. Additionally, traces can now be propagated to the worker nodes as well.
+
+
+## Background
+OpenTelemetry is a powerful serviceability framework that helps to gain insights into the performance and behaviour of the systems. It facilitates generation, collection, and management of telemetry data such as traces.
+
+
+### Existing Open Telemetry
+![Traces existing implementation](/RFC-0011-open-telemetry/traces-existing-implementation-oss-presto.png)
+
+
+The OSS Presto had a basic implementation of Open Telemetry with few limitations listed below.
+- Have only less number of spans and covers only query state transitions like dispatching, planning, analyzing, and
+scheduling. ie. it does not cover most of the internal operations during a query execution.
+- No additional information is being passed as query attributes or events.
+- No correlation between traceId and queryId.
+- Does not identify failed queries
+- No context propagation to track worker nodes.
+
```

Comment:
Here you should point out specifically which SPI methods used for tracing make implementing the particular features difficult